### ISSUE TITLE
Need unit tests

### ISSUE THREAD

**Issue by makkus** [grisu/grisu#54]
This is getting more and more pressing, since more people write code for the core libraries now.

**Comment by makkus** [comment]
Added  notes to test in wiki: https://github.com/grisu/grisu/wiki/Tests

Please state your opinion here if you have one :-)

**Comment by yuriyh** [comment]
integration tests are useful, but I don't think it makes sense to run them on every maven build. It just takes too much time. 

And I think we should have enough coverage for unit tests, at least in the long run.

**Comment by makkus** [comment]
Yeah, I don't want to run them on every maven run, but at maybe for nightly builds and before releases.

How do you propose to create those unit tests, for example for file transfers/job submissions? "Mocking" the grid? Honestly, at the moment I can't really think of a way to do that that would make sense.... Sure there is one, but I can't see it...

**Comment by yuriyh** [comment]
Yeah, mocking. For job submission: need some kind of mock job submitter that will accept jobs and reply with meaningful information on job status query.  

Not sure about filesystems yet. And yes, I know this can take a LOT of time :)

**Comment by makkus** [comment]
Not quite sure whether it makes sense to have this issue in the milestone, it'll take quite a bit of time to have any kind of coverage that makes an impact. I'd guess at least 1 month, probably 2, if I worked on nothing else...

**Comment by makkus** [comment]
Removing from milestone. Too big a ticket to be associated with milestone for now. But useful to have to commit code against.
